### PR TITLE
Add Documentation From Aboutcode

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
This PR Migrates scancode-toolkit documentation from [aboutcode](https://github.com/nexB/aboutcode/tree/master/docs/source/scancode-toolkit) after adding the final improvements/changes via https://github.com/nexB/aboutcode/pull/33 .

[WIP] Can be Merged after these tasks:

- [x] https://github.com/nexB/aboutcode/pull/33 is Merged
- [x] Docs are Migrated
- [ ] Extensive Review of the Documentation

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

### CODE REVIEW

File: docs/source/cli-reference/basic-options.rst [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+        The option ``--license-text-diagnostics`` and ``--is-license-text`` are sub-options of
+        ``--license-text``. ``--is-license-text`` is a Post-Scan Option.
+
+    With the ``--license-text`` option, the scan results attribute "matched text" includes the matched text
+    for the detected license.
+
+    An example Scan::
+
+        scancode -cplieu --json-pp output.json samples --license-text
```
Comment (majurg) [inline docs/source/cli-reference/basic-options.rst]: I think for the basic options documentation, we need to focus on these particular options in this command. 

For example, we should restructure this doc to:
1) tell the user what each of the `-cplieu` flags mean. These are the primary options that people are interested in using
2) tell the user what `--json-pp` means, AND some of the various other output formats like `csv`
3) Move the rest of the option docs to `core-options.rst` OR `extra-options.rst` or similar. 

Basically, we want to have `basic-options.rst` to cover just the basic options that a new user would want to know about before running scancode. More advanced users can look elsewhere for more information.

**Comment by AyanSinhaMahapatra** [comment]
1 Travis Build failed with "The job exceeded the maximum time limit for jobs, and has been terminated". @majurg Does this happen often?

File: docs/source/cli-reference/list-options.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
+.. _cli_list_options:
+
+All Available Options
+=====================
+
+This section contains an exhaustive list of all Scancode options, arranged in various sections.
+The sections are as follows:
+
+- Basic Scan Options
+- Core Scan Options
```
Comment (majurg) [inline docs/source/cli-reference/list-options.rst]: We probably want to rename `Core Scan Options` and its `rst` file to something like `Extra Options` or `Advanced Options` IMO.

File: docs/source/cli-reference/output-filters-and-control.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    ::
+
+        "path": "JGroups/LICENSE"
+
+    .. include::  /rst_snippets/note_snippets/control_strip_full_root.rst
+
+----
+
+``--ignore-author <pattern>`` Option
```
Comment (majurg) [inline docs/source/cli-reference/output-filters-and-control.rst]: We should also document the more popular `--ignore` option here as well.

**Comment by pombredanne** [comment]
@AyanSinhaMahapatra I respun the job. These are totally random flaky failures where the CI flaps for no reason I can fathom.

File: docs/source/cli-reference/list-options.rst (hunk 2 above)
Comment (AyanSinhaMahapatra) [reply docs/source/cli-reference/list-options.rst]: Advanced Options sounds great IMO. Also, note that I have used the structure/grouping of options mentioned in both the help text and in the code `help_group=CORE_GROUP`, it'd be okay if I deviate from that?

File: docs/source/cli-reference/output-filters-and-control.rst (hunk 3 above)
Comment (AyanSinhaMahapatra) [reply docs/source/cli-reference/output-filters-and-control.rst]: So that's grouped under[ `Pre-Scan` Options](https://aboutcode.readthedocs.io/en/latest/scancode-toolkit/cli-reference/list-options.html#all-pre-scan-options). Again, here I'm following the groups of options to be consistent with the help text.

File: docs/source/cli-reference/basic-options.rst (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply docs/source/cli-reference/basic-options.rst]: I'm looking into point `1`.  

And IMO as output-format.rst already has things mentioned in `2`, should I add links to it? 

So what I was trying to achieve through this page structure is grouping command-line options consistent with that of the help text (and groups used while coding). So moving other options (other than `-cplieu`) to another page is still okay?

Since your other comments also raise similar concerns with the grouping of CLI options, I'm starting a discussion @ gitter, in the gsod channel.

Comment (AyanSinhaMahapatra) [reply docs/source/cli-reference/basic-options.rst]: Basically what I mean is, `synopsis.rst` is the page that gives basic users an overview of the options, outputs, and scan command variation. `basic-options.rst` just has the CLI options grouped as basic options.

Comment (majurg) [reply docs/source/cli-reference/basic-options.rst]: Yes, I suppose following the various groups is fine. I had forgotten that this is how they are structured internally in the code itself.

File: docs/source/cli-reference/list-options.rst (hunk 2 above)
Comment (majurg) [reply docs/source/cli-reference/list-options.rst]: Yes, I suppose following the various groups is fine. I had forgotten that this is how they are structured internally in the code itself.

File: docs/source/cli-reference/output-filters-and-control.rst (hunk 3 above)
Comment (majurg) [reply docs/source/cli-reference/output-filters-and-control.rst]: Yes, I suppose following the various groups is fine. I had forgotten that this is how they are structured internally in the code itself.

**Comment by majurg** [comment]
@pombredanne Anything to add here?

**Comment by pombredanne** [comment]
@AyanSinhaMahapatra Thanks. All good and merged!
